internal/fsutil/path_test.go

```diff
@@ -0,0 +1,52 @@
+package fsutil_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/chisel/internal/fsutil"
+)
+
+var dirTests = []struct {
+	input  string
+	output string
+}{
+	{"/a/b/c", "/a/b/"},
+	{"/a/b/c/", "/a/b/"},
+	{"/a/b/c//", "/a/b/"},
+	{"/a/b/c/././", "/a/b/"},
+	{"/a/b/c/.././", "/a/"},
+	{"/a/b//c", "/a/b/"},
+	{"/a/b/./c", "/a/b/"},
+	{"a/b/./c", "a/b/"},
+	{"./a/b/./c", "a/b/"},
+	{"/", "/"},
+	{"///.///", "/"},
+	{".", "./"},
+	{"", "./"},
+}
+
+var cleanTests = []struct {
+	input  string
+	output string
+}{
+	{"/a/b/c", "/a/b/c"},
```

Comment:
I think you should add "/" root here :)

Reply:
Thanks! Indeed it didn't handle `/` correctly. It should be fixed now and there test cases testing various forms of the root directory. Please check it out and resolve if appropriate.